streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/worker/tasks/StreamTask.java

```diff
@@ -5,98 +5,83 @@
-    Map<String, String> queueConf = new HashMap<>();
-    worker.getConfig().forEach((k, v) -> queueConf.put(k, String.valueOf(v)));
-    String queueSize = (String) worker.getConfig()
-        .getOrDefault(Config.CHANNEL_SIZE, Config.CHANNEL_SIZE_DEFAULT);
-    queueConf.put(Config.CHANNEL_SIZE, queueSize);
-    queueConf.put(Config.TASK_JOB_ID, Ray.getRuntimeContext().getCurrentJobId().toString());
-    String channelType = (String) worker.getConfig()
-        .getOrDefault(Config.CHANNEL_TYPE, Config.MEMORY_CHANNEL);
-    queueConf.put(Config.CHANNEL_TYPE, channelType);
-
-    ExecutionGraph executionGraph = worker.getExecutionGraph();
-    ExecutionNode executionNode = worker.getExecutionNode();
-
-    // writers
-    writers = new HashMap<>();
-    List<ExecutionEdge> outputEdges = executionNode.getOutputEdges();
-    List<Collector> collectors = new ArrayList<>();
+    ExecutionVertex executionVertex = jobWorker.getExecutionVertex();
+
+    // consumer
+    List<ExecutionEdge> outputEdges = executionVertex.getOutputEdges();
+    Map<String, ActorId> outputActor = new HashMap<>();
     for (ExecutionEdge edge : outputEdges) {
-      Map<String, ActorId> outputActorIds = new HashMap<>();
-      Map<Integer, RayActor> taskId2Worker = executionGraph
-          .getTaskId2WorkerByNodeId(edge.getTargetNodeId());
-      taskId2Worker.forEach((targetTaskId, targetActor) -> {
-        String queueName = ChannelID.genIdStr(taskId, targetTaskId, executionGraph.getBuildTime());
-        outputActorIds.put(queueName, targetActor.getId());
+      String queueName = ChannelID.genIdStr(
+          taskId, edge.getTargetVertex().getVertexId(), executionVertex.getBuildTime());
+      outputActor.put(queueName, edge.getTargetVertex().getWorkerActorId());
+    }
+    if (!outputActor.isEmpty()) {
+      List<String> channelIDs = new ArrayList<>();
+      List<ActorId> targetActorIds = new ArrayList<>();
+      outputActor.forEach((vertexId, actorId) -> {
+        channelIDs.add(vertexId);
+        targetActorIds.add(actorId);
       });
-
-      if (!outputActorIds.isEmpty()) {
-        List<String> channelIDs = new ArrayList<>();
-        List<ActorId> toActorIds = new ArrayList<>();
-        outputActorIds.forEach((k, v) -> {
-          channelIDs.add(k);
-          toActorIds.add(v);
-        });
-        DataWriter writer = new DataWriter(channelIDs, toActorIds, queueConf);
-        LOG.info("Create DataWriter succeed.");
-        writers.put(edge, writer);
-        Partition partition = edge.getPartition();
-        collectors.add(new OutputCollector(channelIDs, writer, partition));
-      }
+      DataWriter writer = new DataWriter(channelIDs, targetActorIds, jobWorker.getWorkerConfig());
+      collectors.add(new OutputCollector(channelIDs, writer,
+          executionVertex.getOutputEdges().get(0).getPartition()));
     }
 
-    // consumer
-    List<ExecutionEdge> inputEdges = executionNode.getInputsEdges();
+    // producer
```

Comment:
Is there anything I miss? I think it's a `consumer`

Reply:
Thanks. It has been modified.